x-pack/plugins/apm/public/components/app/trace_explorer/index.tsx

```diff
@@ -55,120 +55,99 @@ export function TraceExplorer() {
-  );
+  const routePath = useApmRoutePath();
 
   return (
-    <EuiFlexGroup direction="column" gutterSize="s">
-      <EuiFlexItem>
-        <EuiFlexGroup direction="row">
-          <EuiFlexItem grow>
-            <TraceSearchBox
-              query={query}
-              error={false}
-              loading={false}
-              onQueryCommit={() => {
-                history.push({
-                  ...history.location,
-                  search: fromQuery({
-                    ...toQuery(history.location.search),
-                    query: query.query,
-                    type: query.type,
-                  }),
-                });
-              }}
-              onQueryChange={(nextQuery) => {
-                setQuery(nextQuery);
-              }}
-            />
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <ApmDatePicker />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiFlexItem>
-      <EuiFlexItem>
-        <WaterfallWithSummary
-          waterfallFetchResult={waterfallFetchResult}
-          traceSamples={traceSamplesFetchResult.data?.traceSamples}
-          traceSamplesFetchStatus={traceSamplesFetchResult.status}
-          environment={environment}
-          onSampleClick={(sample) => {
-            push(history, {
-              query: {
-                traceId: sample.traceId,
-                transactionId: sample.transactionId,
-                waterfallItemId: '',
-              },
-            });
-          }}
-          onTabClick={(nextDetailTab) => {
-            push(history, {
-              query: {
-                detailTab: nextDetailTab,
-              },
-            });
-          }}
-          detailTab={detailTab}
-          waterfallItemId={waterfallItemId}
-          serviceName={
-            waterfallFetchResult.waterfall.entryWaterfallTransaction?.doc
-              .service.name
-          }
-          showCriticalPath={showCriticalPath}
-          onShowCriticalPathChange={(nextShowCriticalPath) => {
-            push(history, {
-              query: {
-                showCriticalPath: nextShowCriticalPath ? 'true' : 'false',
-              },
-            });
-          }}
-        />
-      </EuiFlexItem>
-    </EuiFlexGroup>
+    <TraceExplorerSamplesFetcherContextProvider params={params}>
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup direction="row">
+            <EuiFlexItem grow>
+              <TraceSearchBox
+                query={query}
+                error={false}
+                loading={false}
+                onQueryCommit={() => {
+                  history.push({
+                    ...history.location,
+                    search: fromQuery({
+                      ...toQuery(history.location.search),
+                      query: query.query,
+                      type: query.type,
+                    }),
+                  });
+                }}
```

Comment:
Can't you use the [helper push function](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/components/shared/links/url_helpers.ts#L53:L53) instead?

Reply:
Yes, good one. I'm just wrapping old code here but it's a small change to make anyway.